src/Component/Tests/State/Provider/DeserializeProviderTest.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace State\Provider;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Sylius\Resource\Context\Context;
+use Sylius\Resource\Context\Option\RequestOption;
+use Sylius\Resource\Metadata\Delete;
+use Sylius\Resource\Metadata\HttpOperation;
+use Sylius\Resource\Metadata\ResourceMetadata;
+use Sylius\Resource\State\Provider\DeserializeProvider;
+use Sylius\Resource\State\ProviderInterface;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class DeserializeProviderTest extends TestCase
+{
+    use ProphecyTrait;
+
+    /** @var ProviderInterface|ObjectProphecy */
+    private $decorated;
```

Comment:
```suggestion
    private ProviderInterface|ObjectProphecy $decorated;
```

Reply:
@loic425 I think something happened at rebase, property type is not used here and on other properties